### PR TITLE
Post merge cleanup

### LINKED ISSUE #199: Disable internal functions once they're merged into core.
We need to make sure there are no errors for sites that are running our plugin once some of our internal functions are merged into core. To do so, we should depreciate our internal functions and make them wrappers for their counterparts in core when possible to be mindful of back compatibility for sites using our internal functions to build their own markup patterns.

### PR DESCRIPTION
**Pull request by joemcgill**
Move all of the core functions into a new file named `wp-tevko-core-functions.php`
that is only loaded if `wp_get_attachment_image_srcset()` does not exist. Otherwise,
we load a set of compatability shims, `wp-tevko-compat-shims`,  for people using our
core functions in their themes.

Resolves #199

### CODE REVIEW

**Comment by tevko** [comment]
:+1:

**Comment by Shapeshifter3** [comment]
Joe, I have a question:

I'm currently using WordPress 4.31 + the 2.5.2 version on this Plugin. I'm contemplating converting to the nightly build of WordPress, to use the embedded version. Would it be better for me to do that NOW before the plugin update goes to version 3.0.0 OR do it AFTER the update.

Which would leave me with a cleaner install of plugin components?

**Comment by joemcgill** [comment]
Hey shapeshifter! I'd probably suggest waiting until we get version 3.0 out
if you can.

**Comment by Shapeshifter3** [comment]
Oh well,

I got antsy and just uninstalled the plugin, used another plugin to optimize my database, deleted my cache, and then updated to the nightly build.

I'll let you know if I run into any problems (sometimes I just don't have any self-control and get carried away).

Thanks for your quick response!

File: wp-tevko-compat-shims.php [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+ * @deprecated 3.0 Use 'wp_get_attachment_image_srcset()'
+ * @see 'wp_get_attachment_image_srcset()'
+ *
+ * @param int    $id   Image attachment ID.
+ * @param string $size Optional. Name of image size. Default value: 'medium'.
+ * @return string|bool A 'srcset' value string or false.
+ */
+function tevkori_get_srcset( $id, $size = 'medium' ) {
+	_deprecated_function( __FUNCTION__, '3.0.0', 'wp_get_attachment_image_srcset()' );
+	$srcset_array = tevkori_get_srcset_array( $id, $size );
```
Comment (jaspermdegroot) [inline wp-tevko-compat-shims.php]: Shouldn't this be ```return wp_get_attachment_image_srcset( $id, $size );```?

File: wp-tevko-compat-shims.php [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+ * @deprecated 3.0 Use 'wp_get_attachment_image_srcset()'
+ * @see 'wp_get_attachment_image_srcset()'
+ *
+ * @param int    $id   Image attachment ID.
+ * @param string $size Optional. Name of image size. Default value: 'medium'.
+ * @return string|bool A full 'srcset' string or false.
+ */
+function tevkori_get_srcset_string( $id, $size = 'medium' ) {
+	_deprecated_function( __FUNCTION__, '3.0.0', 'wp_get_attachment_image_srcset()' );
+	$srcset_value = tevkori_get_srcset( $id, $size );
```
Comment (jaspermdegroot) [inline wp-tevko-compat-shims.php]: Call ```wp_get_attachment_image_srcset``` instead?

File: wp-tevko-responsive-images.php [hunk 3]
```diff
@@ -48,403 +48,8 @@ function tevkori_get_picturefill() {
...
-				// Set the 'sizes' attribute if sizes were returned.
-				if ( $sizes ) {
-					$attr['sizes'] = $sizes;
-				}
-			}
-		}
-	}
-
-	return $attr;
+if ( ! function_exists( 'wp_get_attachment_image_srcset' ) ) {
```
Comment (jaspermdegroot) [inline wp-tevko-responsive-images.php]: I think we should add an explanation in a comment here for people who didn't check the readme/changelog.

**Comment by jaspermdegroot** [comment]
While reviewing this I noticed that I found our function descriptions a bit confusing sometimes so I created PR https://github.com/ResponsiveImagesCG/wp-tevko-responsive-images/pull/214. If we decide to merge it we should also update the descriptions in this PR.

**Comment by joemcgill** [comment]
Thanks @jaspermdegroot. For the two srcset functions, I left the old logic chain intact in order to preserve any filters someone had applied to `tevkori_srcset_array`. Your function descriptions in #214 were an improvement so I merged and will apply the changes to this PR before we merge.

**Pull request merged**